### PR TITLE
Reuse Uptest workflow and standardize Makefile targets 

### PR DESCRIPTION
**Pull request by turkenh**
### Description of your changes

Consumes reusable workflows introduced in https://github.com/upbound/uptest/pull/39
Also, standardize Makefile targets and cleans up old script.

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

```
export UPTEST_CLOUD_CREDENTIALS=<your-token-here>
UPTEST_EXAMPLE_LIST=examples/iam/user.yaml make e2e
```

I'll only be able to test GitHub actions after merging this PR.

### CODE REVIEW

File: cluster/test/setup.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+set -aeuo pipefail
+
+echo "Running setup.sh"
+echo "Creating cloud credential secret..."
+${KUBECTL} -n upbound-system create secret generic provider-secret --from-literal=credentials="${UPTEST_CLOUD_CREDENTIALS}" --dry-run=client -o yaml | ${KUBECTL} apply -f -
```
Comment (ulucinar) [inline cluster/test/setup.sh:6]: Is there a reason we dry-run and then pipe to an apply instead of running without the `--dry-run` option?

**Review by ulucinar — APPROVED** [review]
Thanks @turkenh, lgtm.

File: cluster/test/setup.sh (hunk 1 above)
Comment (turkenh) [reply cluster/test/setup.sh:6]: yes, if I don't include that it fails when secret exists.

```
kubectl create secret generic aws-secret -n upbound-system --from-file=creds=/Users/hasanturken/.aws/credentials -o yaml | kubectl apply -f -
error: failed to create secret secrets "aws-secret" already exists
error: no objects passed to apply
```

[the remark this answers is not in the document]
Comment (turkenh) [reply cluster/test/setup.sh:6]: `e2e` is expected to run with CI and `uptest` is not different than the previous `uptest-local` target. I'll update the docs, e.g. [this](https://github.com/upbound/upjet/blob/main/docs/testing-instructions.md), accordingly.

[the remark this answers is not in the document]
Comment (turkenh) [reply cluster/test/setup.sh:6]: Just pushed a new commit adding some comments and a deprecation notice for `uptest-local` target.

**Comment by turkenh** [comment]
@ulucinar pushed a commit and responded your comments, could you have another look?

**Review by ulucinar — APPROVED** [review]
Thanks @turkenh, lgtm.

**Pull request merged**